README.md

```diff
@@ -1,86 +1,47 @@
-Adding the `--graph <graph_type>` flag to the end of the command lines above will print out a node-and-edge diagram of the chosen service. 
-Supported values for `<graph_type>` include [mermaid](https://mermaid-js.github.io/) and [dot](https://graphviz.org/doc/info/lang.html).
-
-## Structure of the Template Project
-The `src` directory contains the following files:
-
-```txt
-src/main.rs         - This file contains the `main` function, which handles command-line arguments and launches the appropriate service.
-src/protocol.rs     - This file contains the `Message` enum that defines the messages that can be sent between instances.
-src/helpers.rs      - This file contains helper functions that are invoked from Hydroflow code in multiple services
-src/<role>.rs       - The code for a service with the given role. Default files are provided for `server` and `client`.
+## Viewing Help
+```console
+cargo run -- --help
 ```
 
-The `src/main.rs` file is where the command-line arguments are parsed and the appropriate service is launched.
-It also contains the `Opts` struct, which uses the [clap](https://docs.rs/clap/latest/clap/) crate to 
-specify the command-line arguments that are accepted by the application.
-It is possible to change the command-line arguments by modifying the `Opts` struct.
-Before launching the service, the `main` function binds to the specified address and prints out the address that was bound to.
-
-The `src/protocol` file contains the enum `Message`, which can include messages with very different structures. 
-The `Message` must provide the `Serialize` and `Deserialize` traits, which are used by the [serde](https://docs.serde.rs/serde/) crate.
-In the application, the `Message` enum includes an `Echo` message that has a (`String`) payload and timestamp; it also includes
-`Heartbeat` and `HeartbeatAck` messages that carry no information other than their type. (Messages are delivered with the sender 
-address attached, so these empty message types can be useful.) 
-
-The `src/helpers.rs` file contains any helper functions that are needed. In our example this is just a function to print
-a dataflow graph representation of the hydroflow code to stdout.
-
-Each service file comes with a skeleton Hydroflow spec that provides an inbound communication channel and an outbound communication channel, both 
-bound to the specified address and port. The channels are named `inbound_chan` and `outbound_chan`, and are accessed in hydroflow code using the 
-`source_serde` and `sink_serde` operators respectively. The single address/port pair is sufficient in general to support multiple
-different `Message` types across multiple services and instances. Upon receipt, messages are handled by the appropriate code using hydroflow's `demux`
-operator. It is also possible to open more channels to segregate traffic to different IP addresses or ports, simply by copying the 
-patterns that define and use the `inbound` and `outbound` channels.
```

Comment:
I have inlined most of this within the code itself. I am hoping it becomes easier to read it at the same site as the code itself.